Assets/LeapMotion/Core/Scripts/Attributes/CurveBounds.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+
+namespace Leap.Unity.Attributes {
+
+  /// <summary>
+  /// You can use this attribute to mark that an AnimationCurve can only have
+  /// values that fall within specific bounds.  The user will be prevented from
+  /// entering a curve that lies outside of these bounds.
+  /// </summary>
+  public class CurveBounds : CombinablePropertyAttribute, IFullPropertyDrawer {
+    public readonly Rect bounds;
+    public readonly Color color;
+
+    public CurveBounds(Rect bounds) {
+      this.bounds = bounds;
+      color = Color.green;
+    }
+
+    public CurveBounds(Rect bounds, Color color) {
+      this.bounds = bounds;
+      this.color = color;
+    }
+
+    public CurveBounds(float width, float height) {
+      bounds = new Rect(0, 0, width, height);
+      color = Color.green;
+    }
+
+    public CurveBounds(float width, float height, Color color) {
```

Comment:
I'm getting a compile error when I try to do
```
  [CurveBounds(2f, 2f, Color.blue)]
  public AnimationCurve arbitraryBoundsCurve;
```
```
Assets/CurveBoundsTest.cs(8,30): error CS0181: Attribute constructor parameter has type `UnityEngine.Color', which is not a valid attribute parameter type
```

Reply:
OMG I never new this... 
> Only primitive constants or arrays of primitives can be used as attribute parameters.
This is crazy!  Ok well i guess I could either remove color or have users input RGB independently, what do you think?